src/scripts/main.js

```diff
@@ -1,3 +1,335 @@
 'use strict';
 
-// write your code here
+const game = document.querySelector('.container');
+
+const gameHeader = game.querySelector('.game-header');
+const gameMessageContainer = game.querySelector('.message-container');
+
+const gameButonStart = gameHeader.querySelector('.start');
+const gameScore = game.querySelector('.game-score');
+
+const fieldCell = game.querySelectorAll('.field-cell');
+
+const cell = [...fieldCell];
+
+const messageLose = gameMessageContainer.querySelector('.message-lose');
+const messageWin = gameMessageContainer.querySelector('.message-win');
+const messageStart = gameMessageContainer.querySelector('.message-start');
+
+function removeEventHandler() {
+  document.removeEventListener('keydown', keydownEventHandler);
+}
+
+function addEventHandler() {
+  document.addEventListener('keydown', keydownEventHandler);
+}
+
+const Directions = {
+  ArrowUp: 'up',
+  ArrowDown: 'down',
+  ArrowLeft: 'left',
+  ArrowRight: 'right'
+};
+
+function keydownEventHandler(el) {
+  const direction = Directions[el.key];
+
+  if (direction) {
+    addRandomNumber();
+    move(direction);
+    removeClas();
+    win();
+    isGameOver();
+    gameScore.textContent = score;
+  }
+}
+
+function move(direction) {
+  switch (direction) {
+    case 'up':
+      moveUp();
+      break;
+    case 'down':
+      moveDown();
+      break;
+    case 'left':
+      moveLeft();
+      break;
+    case 'right':
+      moveRight();
+      break;
+  }
+}
+
+
+removeEventHandler();
+
+let score = 0;
+
+function removeClas() {
+  return [...fieldCell].map(el => {
+    if (el.textContent) {
+      el.classList.remove(...el.classList);
+      el.classList.add('field-cell');
+      el.classList.add(`field-cell--${el.textContent}`);
+    } else {
+      if (el.classList.contains('field-cell')) {
+        el.classList.remove(...el.classList);
+        el.classList.add('field-cell');
+      }
+    }
+  });
+}
+
+function generateRandomNumber() {
+  const randomNumber = Math.random();
+
+  if (randomNumber < 0.1) {
+    return 4;
+  } else if (randomNumber < 0.9) {
+    return 2;
+  } else {
+    return 2;
```

Comment:
```suggestion
  if (randomNumber < 0.1) {
    return 4;
  }
  
  if (randomNumber < 0.9) {
    return 2;
  }
  
  return 2;
```
There is no need to use if...else. Code will not go further after `return` anyway